### PR TITLE
[1.19.3] Add option to completely hide a crash-callable depending on a runtime value

### PR DESCRIPTION
**Pull request by XFactHD**
This PR adds the ability to completely hide a crash-callable depending on a dynamic runtime value such as a config setting.

This allows mods to disable warnings which for example only apply when a certain gameplay feature is enabled via config setting. At the moment this requires printing some kind of placeholder text when the warning doesn't apply, which tends to confuse users.

Example of such a use case: https://github.com/XFactHD/FramedBlocks/blob/1.19.3/src/main/java/xfacthd/framedblocks/FramedBlocks.java#L92-L109
This would print `FramedBlocks BlockEntity Warning: Not applicable` to the crash report, which is unnecessarily confusing when there is nothing to warn the user about.

### CODE REVIEW

File: src/test/java/net/minecraftforge/debug/CrashCallableTest.java
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/debug/CrashCallableTest.java:23]: This could mention the `F3 + C` shortcut which force-crashes the game when held for 10 seconds, for ease of testing. I'll leave it up to you whether to add it or not.

File: fmlcore/src/main/java/net/minecraftforge/fml/CrashReportCallables.java
Comment (sciwhiz12) [inline fmlcore/src/main/java/net/minecraftforge/fml/CrashReportCallables.java:93]: I'm pondering whether we should take this out of the public API (with `@ApiStatus.Internal`). I'll leave that decision to the core team for now.

**Review by sciwhiz12 — APPROVED** [review]
Works in testing using `F3 + C`.

File: src/test/java/net/minecraftforge/debug/CrashCallableTest.java
Comment (XFactHD) [reply src/test/java/net/minecraftforge/debug/CrashCallableTest.java:23]: Added a comment mentioning the shortcut

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**
